ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1518,20 +1518,24 @@ protected void maybeProcessCallbacks() {
                 blobInfo = new BlobInfo(serverBlobProperties, result.getDecryptedUserMetadata().array(), lifeVersion);
               }
               ByteBuf decryptedBlobContent = result.getDecryptedBlobContent();
-              totalSize = decryptedBlobContent.readableBytes();
-              if (!resolveRange(totalSize)) {
-                routerMetrics.getFirstDataChunkLatencyMs.update(System.currentTimeMillis() - submissionTimeMs);
-                routerMetrics.getDataChunkLatencyMs.update(System.currentTimeMillis() - initializedTimeMs);
-                ByteBuf decompressedContent = decompressContent(decryptedBlobContent);
-                if (decompressedContent != null) {
+              routerMetrics.getFirstDataChunkLatencyMs.update(System.currentTimeMillis() - submissionTimeMs);
+              routerMetrics.getDataChunkLatencyMs.update(System.currentTimeMillis() - initializedTimeMs);
+              ByteBuf decompressedContent = decompressContent(decryptedBlobContent);
+              if (decompressedContent == null) {
```

Comment:
nit: Can u put the most likely case first to avoid a code jump ? decompressedContent != null first.